src/click/_termui_impl.py

```diff
@@ -424,26 +424,33 @@ def _pipepager(generator: cabc.Iterable[str], cmd: str, color: bool | None) -> N
                 text = strip_ansi(text)
 
             stdin.write(text.encode(encoding, "replace"))
-    except (OSError, KeyboardInterrupt):
-        pass
-    else:
+            stdin.flush()
```

Comment:
Close implies flush, so this one is not necessary. https://stackoverflow.com/questions/2447143/does-close-imply-flush-in-python
```suggestion
```